src/node_contextify.cc

```diff
@@ -110,9 +110,28 @@ using v8::Value;
 
 namespace {
 
+// Utility function to get the file extension from a filename
+std::string GetFileExtension(std::string_view filename) {
+  size_t dot_position = filename.find_last_of('.');
+  if (dot_position == std::string_view::npos) {
+    return "";  // No extension found
+  }
+  return std::string(filename.substr(dot_position));
+}
+
+std::string GetWarningMessage(const std::string& file_extension) {
+  if (file_extension == ".cjs") {
+    return "Cannot use 'import' statement in a .cjs file. "
+           "CommonJS files (.cjs) do not support 'import'. Use 'require()' instead.";
+  }
+  return "To load an ES module, set \"type\": \"module\" in the package.json "
+         "or use the .mjs extension.";
+}
+
 // Convert an int to a V8 Name (String or Symbol).
 Local<Name> Uint32ToName(Local<Context> context, uint32_t index) {
-  return Uint32::New(context->GetIsolate(), index)->ToString(context)
+  return Uint32::New(context->GetIsolate(), index)
```

Comment:
Can you revert the unrelated formatting changes? These would make backporting harder.